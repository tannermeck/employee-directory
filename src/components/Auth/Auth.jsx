export default function Auth(){
    return (
        <fieldset>
            <legend>Sign-in</legend>
                <form>
                    <label htmlFor='email'>Email:</label>
                    <input type='email' id='email' />
                    <label htmlFor='password'>Password:</label>
                    <input type='password' id='password' />
                    <button type='submit'>Sign-in</button>
                </form>
        </fieldset>
    )
}