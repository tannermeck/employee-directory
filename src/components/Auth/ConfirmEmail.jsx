import style from './confirmEmail.css';

export default function ConfirmEmail(){
    return (
        <section className={style.headings}>
          <h1>Registration Successful!</h1>
          <p>
            Please check your email in order to confirm your account and sign in.
          </p>
        </section>
      );
}