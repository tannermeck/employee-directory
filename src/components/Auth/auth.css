.authContainer {
    color: white;
    width: fit-content;
    margin: auto;
    margin-top: 30px;
    border-color: rgb(255, 208, 0);
}
legend {
    text-align: center;
    color: rgb(255, 208, 0);
}
.formContainer {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}
label {
    font-size: 1rem;
    padding: 10px;
    width: 365px;
    text-align: left;
}
input {
    width: 350px;
    margin: 10px;
    padding: 5px;
    border: 1px solid white;
    background-color: black;
}
button {
    width: 365px;
    margin-top: 10px;
    font-size: 20px;
    font-family: 'Times New Roman', Times, serif;
    height: 40px;
    background-color: rgb(255, 208, 0);
}
button:hover {
    cursor: pointer;
}
h3 {
    color: white;
    text-align: center;
    font-size: 1.5rem;
}
.register {
    color: white;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
}
.register a {
    color:rgb(255, 208, 0);
    text-decoration: underline;
    padding-left: 5px;
}
.passwordInput {
    color: white;
}