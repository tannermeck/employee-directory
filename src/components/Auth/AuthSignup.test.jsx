import {render} from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import Auth from './Auth';
import { UserProvider } from '../../context/UserContext/UserContext';

jest.mock('../../services/users');
jest.mock('../../context/UserContext/UserContext');

it('should render a signup screen', () => {
    const {container} = render(
        <UserProvider>
            <MemoryRouter>
                <Auth newUser />
            </MemoryRouter>
        </UserProvider>
    )
    expect(container).toMatchSnapshot()
})