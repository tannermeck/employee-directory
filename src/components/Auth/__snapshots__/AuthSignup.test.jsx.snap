// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a signup screen 1`] = `
<div>
  <h3>
    Welcome
  </h3>
  <fieldset
    class="authContainer"
  >
    <legend>
      SIGN UP
    </legend>
    <form
      class="formContainer"
    >
      <label
        for="email"
      >
        Email:
      </label>
      <input
        class="passwordInput"
        id="email"
        type="email"
        value=""
      />
      <label
        for="password"
      >
        Password:
      </label>
      <input
        class="passwordInput"
        id="password"
        type="password"
        value=""
      />
      <button
        type="submit"
      >
        SIGN UP
      </button>
      <p />
    </form>
  </fieldset>
  <div>
    <div
      class="register"
    >
      <p>
        Already have an account?
      </p>
      <a
        href="/login"
      >
        Sign In
      </a>
    </div>
  </div>
</div>
`;
